app/src/main/java/doit/study/droid/QuestionsActivity.java

```diff
@@ -24,7 +30,23 @@ protected void onCreate(Bundle savedInstanceState) {
         pagerTabStrip.setNonPrimaryAlpha(0);
         pagerTabStrip.setTabIndicatorColor(0x000000);
         GlobalData gd = (GlobalData) getApplication();
-        mPagerAdapter = new QuestionsPagerAdapter(getSupportFragmentManager(), gd.getQuizData());
+        List<Integer> questionIds = (List<Integer>) gd.retrieve("questionIds");
+        Collections.shuffle(questionIds);
+        if (questionIds.size() > 10) {
+            questionIds = questionIds.subList(0, 10);
+        }
+        AnswerCheckListener answerCheckListener = new AnswerCheckListener() {
+            DatabaseHelper mDBHelper = new DatabaseHelper(QuestionsActivity.this.getApplicationContext());
+            @Override
+            public void onAnswer(int questionId, boolean isRight) {
+                mDBHelper.addStats(questionId, isRight);
+            }
+        };
+        mPagerAdapter = new QuestionsPagerAdapter(
+                getSupportFragmentManager(),
+                answerCheckListener,
```

Comment:
The standard way is force a host activity to implement some interface, than use a callback from a guessed fragment.